### PR TITLE
Fix C# string.IsAbsPath()

### LINKED ISSUE #42250: C# `IsAbsPath` behaves differently from `is_abs_path`
**Godot version:**
v3.2.3.stable.mono.official

**OS/device including version:**
Windows 10

**Issue description:**
C#:
```cs
var path = "res://a/b/c.txt";
var path2 = "D:/a/b/c.txt";
var path3 = "D:\\a\\b\\c.txt";

GD.Print($"cs {path} {path.IsAbsPath()}"); // false
GD.Print($"cs {path2} {path2.IsAbsPath()}"); // true
GD.Print($"cs {path3} {path3.IsAbsPath()}"); // true
```
GDScript:
```gdscript
var path = "res://a/b/c.txt"
var path2 = "D:/a/b/c.txt"
var path3 = "D:\\a\\b\\c.txt"

print("gd %s %s" % [path, path.is_abs_path()]) # true
print("gd %s %s" % [path2, path2.is_abs_path()]) # true
print("gd %s %s" % [path3, path3.is_abs_path()]) # true
```
I see `string.IsAbsPath()` calls `System.IO.Path.IsPathRooted(string)` ~~so the results might be platform-related?~~

Edit: I made mistakes for test code before. It seems `IsPathRooted` won't recognize `res://` as valid drive:
https://github.com/dotnet/runtime/blob/436ddc2aded81547e8479c8ccf2fe6f36db1b82e/src/libraries/System.Private.CoreLib/src/System/IO/Path.Windows.cs#L198

**Steps to reproduce:**
**Minimal reproduction project:**
[IsAbsPath-bug.zip](https://github.com/godotengine/godot/files/5262804/IsAbsPath-bug.zip)

### PR DESCRIPTION
**Pull request by zaevi**
fix #42250

### CODE REVIEW

File: modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs [hunk 1]
```diff
@@ -440,15 +440,20 @@ public static string Insert(this string instance, int pos, string what)
         // </summary>
         public static bool IsAbsPath(this string instance)
         {
-            return System.IO.Path.IsPathRooted(instance);
+            if(string.IsNullOrEmpty(instance))
```
Comment (aaronfranke) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs:443]: Formatting is wrong, `if(` should be `if (`.

**Review by aaronfranke — CHANGES_REQUESTED** [review]
This code should be changed to match the C++ implementation: https://github.com/godotengine/godot/blob/master/core/ustring.cpp#L3670

**Review by neikeq — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.4.

**Pull request merged**